tests/ast/README.md

```diff
@@ -0,0 +1,38 @@
+## AST Tests
+
+This driver enables testing of the AST generation that will be used in the server-side Snowpark implementation, starting with Phase 0.
+
+All generated AST should be tested using this mechanism. To add a test, create a new file under `tests/ast/data`. Files look like the following example. The test driver sets up the session and looks at the accumulated lazy values in the resulting environment.
+
+N.B. No eager evaluation is permitted, as any intermediate batches will not be observed. This can easily be changed if necessary, however.
+
+```python
+## TEST CASE
+
+df = session.table(tables.table1)
+df = df.filter("STR LIKE '%e%'")
+
+## EXPECTED ENCODED AST
+
+[...]
+
+## EXPECTED UNPARSER OUTPUT
+
+res1 = session.table('table1')
+
+res2 = res1.filter('STR LIKE '%e%'')
+```
+
+To generate the expected output the first time the test is run, or when the AST generation changes, run:
+```bash
+pytest --update-expectations tests/ast
+```
+
+For these tests to work, the Unparser must be built in the monorepo:
+```bash
+cd my-monorepo-path
+cd Snowflake/unparser
+sbt assembly
+```
+
```

Comment:
maybe let's mention the update local scripts? We could potentially check them in with this PR as well. I am referring to `src/snowflake/snowpark/_internal/proto/update-from-local.sh` here.

Reply:
proto related staff can be merged separately. Let me take a look based on our discussion on other thread.